File: apps/backend/app/errors/validation_error.py
class ValidationError(Exception):
    '''
    バリデーションエラークラス
    '''
    def __init__(self, error_message: str, field_name: str | None = None):
        '''
        コンストラクタ
        :param error_massage: str: エラーメッセージ
        :param field_name: str: 項目名
        '''
        self.error_message = error_message
        self.field_name = field_name

        super().__init__(error_message)

File: apps/frontend/src/shared/models/validationError.ts
import { FieldValues, FieldPath } from 'react-hook-form';

/**
 * バリデーションエラー情報
 * @param message エラーメッセージ
 * @param fieldname エラーが発生したフィールド名
 */
class ValidationError<TFieldValues extends FieldValues = FieldValues> extends Error {
  message: string;
  fieldname: FieldPath<TFieldValues>;

  constructor(message: string, fieldname: string) {
    super(message);
    this.message = message;
    this.fieldname = fieldname as FieldPath<TFieldValues>;
  }
}
export default ValidationError;
